doc/source/whatsnew/v0.18.1.txt

```diff
@@ -75,6 +75,43 @@ API changes
 
 
 
+- ``pandas.parser.OverflowError`` has been removed and has been replaced with Python's built-in ``OverflowError`` (:issue:`12551`)
```

Comment:
Hmmm...not really sure how to illustrate it.  Even the test that's supposed to get the error thrown actually throws the _real_ OverflowError now.